### ISSUE TITLE
You need to use functions that are compatible with numpy

### ISSUE THREAD

**Issue by mikelane** [python-neural-network/neural-network#3]
The sigmoid code [here](https://github.com/python-neural-network/neural-network/blob/4e7a529c37bce3631251cadb2957532d488181e4/functions.py#L5) will cause the entire neural network to be terrifically slow. I suggest changing it to either

```python
def sigmoid(arr: ndarray) -> ndarray:
    return scipy.expit(arr)
```
or
```python
def sigmoid(arr: ndarray) -> ndarray:
    return 1 / (1 + np.exp(-arr))
```

**Comment by loucadufault** [comment]
first time seeing function annotations in the wild

Few questions:
-Should speed be a concern when first building this, couldn't we always optimize down the line? 
Im only saying this because the main concepts are hard enough to understand as is without even worrying about optimization. 

-From the expit documentation, it says that it applies the expit function element wise to the input ndarray (docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.special.expit.html) how is this better than looping over it ourselves?

FYI for anyone reading this, ndarray is pretty much the same as numpy.array(). ndarray is technically the type, and numpy.array() is the function to build arrays.

**Comment by mikelane** [comment]
> Should speed be a concern when first building this
Absolutely. This isn't a premature optimization, this will make the neural network thousands of times faster. 

> how is this better than looping over it ourselves?
1) Because the looping happens in optimized, compiled C code and 2) because we don't actually have to write the loops themselves.
